server/src/main/java/org/elasticsearch/rest/action/RestChunkedToXContentListener.java

```diff
@@ -37,10 +38,17 @@ public RestChunkedToXContentListener(RestChannel channel, ToXContent.Params para
     @Override
     protected void processResponse(Response response) throws IOException {
         channel.sendResponse(
-            RestResponse.chunked(getRestStatus(response), ChunkedRestResponseBody.fromXContent(response, params, channel, null))
+            RestResponse.chunked(
+                getRestStatus(response),
+                ChunkedRestResponseBody.fromXContent(response, params, channel, releasableFromResponse(response))
+            )
         );
     }
 
+    protected Releasable releasableFromResponse(Response response) {
```

Comment:
Not in love with this approach, but I didn't want to duplicate this across both rest actions and I expect us to need this elsewhere as well eventually.

Reply:
Eh seems ok to me. The "proper" approach would be to make `ChunkedToXContent extends RefCounted` too and then add an `incRef`/`decRef` pair here, but that seems excessive to me for now.